src/MongoDB.Driver.Core/Core/Bindings/BindingHelper.cs

```diff
@@ -0,0 +1,64 @@
+/* Copyright 2019–present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System.Threading;
+using System.Threading.Tasks;
+using MongoDB.Driver.Core.Bindings;
+using MongoDB.Driver.Core.Clusters;
+using MongoDB.Driver.Core.Clusters.ServerSelectors;
+using MongoDB.Driver.Core.Servers;
+
+namespace MongoDB.Driver.Core.Core.Bindings
+{
+    /// <summary>
+    /// This class contains helper methods for the *Binding series of classes
+    /// </summary>
+    internal sealed class BindingHelper
+    {
+        public static bool InShardedTransaction(ICoreSessionHandle session, ICluster cluster) =>
+            session != null && session.IsInTransaction && cluster.Description.Type == ClusterType.Sharded;
+
+        public static IServer SelectServer(
+            ICoreSessionHandle session,
+            ICluster cluster,
+            IServerSelector selector,
+            CancellationToken cancellationToken)
```

Comment:
Let's make the cluster the first parameter. The general rule of parameter ordering is to put the most general parameters ahead of the more specific. Cluster is the most globally scoped of the parameters.

I'd put session second and leave the rest as is.